src/env.h

```diff
@@ -255,6 +255,7 @@ class Environment {
     inline uint32_t* fields();
     inline int fields_count() const;
     inline bool call_init_hook();
+    bool enabled;
```

Comment:
Style: at the very least, rename it to `enabled_`, make it private and only accessible through a getter and setter.  The getter should be const.

That said, I believe AsyncHooks is exposed to JS land.  In that case, enabled should be a new element in the fields_ array.